R/RcppExports.R

```diff
@@ -59,14 +16,55 @@ final_size_cpp <- function(r0, contact_matrix, demography_vector, prop_initial_i
 #' group $i$ across risk groups, and each row *must sum to 1.0*.
 #' @param susceptibility A matrix giving the susceptibility of individuals in
 #' demographic group $i$ and risk group $j$.
-#' @param solver Which solver to use. Options are "iterative" or "newton", for
-#' the iterative solver, or the Newton solver, respectively. Special conditions
-#' apply when using the Newton solver.
-#' @param control A list of named solver options, see *Solver options*.
 #'
-#' @keywords epidemic model
-#' @export
-final_size_grps_cpp <- function(contact_matrix, demography_vector, p_susceptibility, susceptibility, solver, control) {
-    .Call('_finalsize_final_size_grps_cpp', PACKAGE = 'finalsize', contact_matrix, demography_vector, p_susceptibility, susceptibility, solver, control)
+#' @return A list object with named elements: "contact_matrix",
+#' "demography_vector", "p_susceptibility_", and "susceptibility".
+#' The contact matrix is replicated row and column wise for each risk group
+#' and the demography vector is replicated for each risk group.
+.epi_spread <- function(contact_matrix, demography_vector, p_susceptibility, susceptibility) {
+    .Call('_finalsize_epi_spread', PACKAGE = 'finalsize', contact_matrix, demography_vector, p_susceptibility, susceptibility)
+}
```

Comment:
I believe this should remain an R function. Below some benchmarks where `final_size_cpp()` uses `.epi_spread()` (C++ function), and `final_size()` uses the R `epi_spread()` function which is currently on `main`:

|expr                                                                                                                                                                          |      min|       lq|     mean|   median|       uq|       max| neval|
|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------:|--------:|--------:|--------:|--------:|---------:|-----:|
|final_size(contact_matrix = r0 * c_matrix, demography_vector = d_vector,      p_susceptibility = psusc, susceptibility = susc, solver = "newton",      control = control)     | 1.196803| 1.268342| 1.344008| 1.299141| 1.332368|  14.04545| 1e+05|
|final_size_cpp(contact_matrix = r0 * c_matrix, demography_vector = d_vector,      p_susceptibility = psusc, susceptibility = susc, solver = "newton",      control = control) | 1.164557| 1.211472| 1.292458| 1.240458| 1.278536| 215.94878| 1e+05|

Reply:
Thanks, this function is now removed entirely per @TimTaylor's suggestion and is included in the body of `final_size`, see https://github.com/epiverse-trace/finalsize/pull/73/commits/b48f8d859b516acb5b66823b3ae7b9145602f368